transport/ice/IceSession.cpp

```diff
@@ -1091,31 +1270,172 @@ void IceSession::stateCheck(const uint64_t now)
 
 void IceSession::freezePendingProbes(uint64_t now)
 {
-    for (auto it = _candidatePairs.rbegin(); it != _candidatePairs.rend(); ++it)
+    // We are going to clear all remote candidates without candidatePairs
+    _remoteCandidates.clear();
+    auto itEnd = _candidatePairs.end();
+
+    for (auto it = _candidatePairs.begin(); it != itEnd;)
     {
         auto* candidatePair = it->get();
-        if (!candidatePair->nominated && candidatePair->state <= ProbeState::Succeeded)
+        if (candidatePair->nominated)
+        {
+            addCandidateToListIfNotPresent(_remoteCandidates, candidatePair->remoteCandidate);
+            ++it;
+            continue;
+        }
+
+        if (candidatePair->state <= ProbeState::Succeeded)
         {
             candidatePair->freeze();
             if (candidatePair->localCandidate.transportType == TransportType::TCP)
             {
-                if (_eventSink)
-                {
-                    _eventSink->onIceDiscardCandidate(this,
-                        candidatePair->localEndpoint.endpoint,
-                        candidatePair->remoteCandidate.address);
-                }
-                removeRemoteCandidate(candidatePair->remoteCandidate);
-                removeCandidatePair(candidatePair);
-            }
-            else if (candidatePair->remoteCandidate.type == IceCandidate::Type::PRFLX)
-            {
-                // port may be recycled in firewall anyway
                 candidatePair->state = ProbeState::Failed;
-                removeRemoteCandidate(candidatePair->remoteCandidate);
+                onCandidatePairRemoved(candidatePair);
+                std::iter_swap(it, --itEnd);
+                continue;
             }
         }
+        // If didn't fail, then it can be a backup in case the nominated becomes unreachable
+        else if (candidatePair->remoteCandidate.type == IceCandidate::Type::PRFLX &&
```

Comment:
We do remove PRFLX candidates if they FAILED. Probably we could delete SRFLX and HOST too. If they fail, there is low chance to them to start be used later on